zdm/tests/mc_statistics.py

```diff
@@ -0,0 +1,386 @@
+		pwidths,pprobs=survey.make_widths(ICS,Wlogmean,Wlogsigma,Wbins,scale=Wscale)
+		efficiencies=ICS.get_efficiency_from_wlist(dmvals,pwidths,pprobs)
+		
+		# load Parkes data
+		pks=survey.survey()
+		pks.process_survey_file(sdir+'parkes_mb_class_I_and_II.dat')
+		pks.init_DMEG(DMhalo)
+		pks.init_beam(nbins=Nbeams[2],method=2,plot=False,thresh=thresh) # need more bins for Parkes!
+		pwidths,pprobs=survey.make_widths(pks,Wlogmean,Wlogsigma,Wbins,scale=Wscale)
+		efficiencies=pks.get_efficiency_from_wlist(dmvals,pwidths,pprobs)
+		
+		
+		names=['ASKAP/FE','ASKAP/ICS','Parkes/Mb']
+		
+		surveys=[lat50,ICS,pks]
+		if not os.path.isdir('Pickle'):
+			os.mkdir('Pickle')
+		with open('Pickle/'+sprefix+'surveys.pkl', 'wb') as output:
+			pickle.dump(surveys, output, pickle.HIGHEST_PROTOCOL)
+			pickle.dump(names, output, pickle.HIGHEST_PROTOCOL)
+	else:
+		with open('Pickle/'+sprefix+'surveys.pkl', 'rb') as infile:
+			surveys=pickle.load(infile)
+			names=pickle.load(infile)
+			lat50=surveys[0]
+			ICS=surveys[1]
+			pks=surveys[2]
+	print("Initialised surveys ",names)
+	
+	dirnames=['ASKAP_FE','ASKAP_ICS','Parkes_Mb']
+	
+	#### these are hard-coded best-fit parameters ####
+	# initial parameter values. SHOULD BE LOGSIGMA 0.75! (WAS 0.25!?!?!?)
+	# Best-fit parameter values (result from cube iteration)
+	lEmin=30. # log10 in erg
+	lEmax=41.84 # log10 in erg
+	alpha=1.54 # spectral index. WARNING: here F(nu)~nu^-alpha in the code, opposite to the paper!
+	gamma=-1.16 # slope of luminosity distribution function
+	sfr_n=1.77 #scaling with star-formation rate
+	lmean=2.16 # log10 mean of DM host contribution in pc cm^-3
+	lsigma=0.51 # log10 sigma of DM host contribution in pc cm^-3
+	C=4.19 # log10 constant in number per Gpc^-3 yr^-1 at z=0
+	pset=[lEmin,lEmax,alpha,gamma,sfr_n,lmean,lsigma,C]
+	
+	
+	# generates zdm grids for the specified parameter set
+	NewGrids=False
+	if sprefix=='Full':
+		gprefix='best'
+	elif sprefix=='Std':
+		gprefix='Std_best'
+	
+	if NewGrids:
+		print("Generating new grids, set NewGrids=False to save time later")
+		grids=misc_functions.initialise_grids(surveys,zDMgrid, zvals,dmvals,pset,wdist=True,source_evolution=source_evolution,alpha_method=alpha_method)
+		with open('Pickle/'+gprefix+'grids.pkl', 'wb') as output:
+			pickle.dump(grids, output, pickle.HIGHEST_PROTOCOL)
+	else:
+		print("Loading grid ",'Pickle/'+gprefix+'grids.pkl')
+		with open('Pickle/'+gprefix+'grids.pkl', 'rb') as infile:
+			grids=pickle.load(infile)
+	glat50=grids[0]
+	gICS=grids[1]
+	gpks=grids[2]
+	print("Initialised grids")
+	
+	
+	#testing_MC!!! Generate pseudo samples from lat50
+	which=0
+	g=grids[which]
+	s=surveys[which]
+	
+	savefile='mc_sample.npy'
+	
+	try:
+		sample=np.load(savefile)
+		print("Loading ",sample.shape[0]," samples from file ",savefile)
+	except:
+		Nsamples=10000
+		print("Generating ",Nsamples," samples from survey/grid ",which)
+		sample=g.GenMCSample(Nsamples)
```

Comment:
@cwjames1983 I can't seem to find where this new GenMCSample() function is defined. I thought it would be in the zdm.py file which initialises the grid, but it isn't there.

Reply:
Wait it's not? It is on the local version I pushed! OK, I will see what the issue is.